artichoke-backend/src/extn/core/time/args.rs

```diff
@@ -300,6 +334,76 @@ mod tests {
         assert_eq!(0, result.nanoseconds);
     }
 
+    #[test]
+    fn month_supports_string_values() {
+        let mut interp = interpreter();
+
+        let table = [
+            (b"[2022, \"jan\"]", 1),
+            (b"[2022, \"feb\"]", 2),
+            (b"[2022, \"mar\"]", 3),
+            (b"[2022, \"apr\"]", 4),
+            (b"[2022, \"may\"]", 5),
+            (b"[2022, \"jun\"]", 6),
+            (b"[2022, \"jul\"]", 7),
+            (b"[2022, \"aug\"]", 8),
+            (b"[2022, \"sep\"]", 9),
+            (b"[2022, \"oct\"]", 10),
+            (b"[2022, \"nov\"]", 11),
+            (b"[2022, \"dec\"]", 12),
+        ];
+
+        for (input, expected_month) in table {
+            let args = interp.eval(input).unwrap();
+            let mut ary_args: Vec<Value> = interp.try_convert_mut(args).unwrap();
+            let result: Args = interp.try_convert_mut(ary_args.as_mut_slice()).unwrap();
+
+            assert_eq!(expected_month, result.month);
+        }
+    }
+
+    #[test]
+    fn month_supports_string_like_values() {
+        let mut interp = interpreter();
+
+        let args = interp
+            .eval(b"class A; def to_str; \"feb\"; end; end; [2022, A.new]")
```

Comment:
we can make the ruby code look like Ruby code here by using different quotes.

```suggestion
            .eval(b"class A; def to_str; 'feb'; end; end; [2022, A.new]")
```